graph/patterns/alternate-key.md

```diff
@@ -0,0 +1,48 @@
+# Alternate Key Pattern
```

Comment:
I have a general editorial comment (maybe more for @OlgaPodo). 
We should use consistent language across these topics.

- resource vs entity?
- property vs attribute?
- Should we be talking about/mentioning OData?

Reply:
IMO we should use resource as it's more generic than an entity and consistent with JSON representation which is an ultimate target of our design. We also should stick with properties as structural elements of a resource for consistency with other public documentation https://docs.microsoft.com/en-us/graph/api/resources/user?view=graph-rest-1.0#properties ( i see that it's not quite consistent with JSON spec) and leave attributes for metadata description.
I don't think that we need to mention OData in this context. In some patterns Type hierarchy, it's unavoidable.

Reply:
* It should be `resource`. Wherever I wrote entity, it was a blunder. a3966c7
* It should be `attribute`. I will change this if you folks want, but that definition is incorrect
* That is an interesting question. Graph is very heavily reliant (unfortunately) on oData. I would see the case that we have this pattern here and then the workload developers might be like "but that is what oData says anyway, why specify it again here?". I think that saying in this document "hey what we are saying is that we embrace oData" makes it a little bit redundant, but I would say less confusing.

I am open to suggestions though!

Reply:
@XVincentX In the context of Microsoft internal and external documentation when you refer to a resource,  not JSON object, using attribute will be very confusing because Property is used in documentation, DSL/CSDL, and MS REST API guidelines(https://github.com/microsoft/api-guidelines/blob/graph/Guidelines.md). We are creating these patterns for the MS ecosystem and developers so i think it should be consistent.

Reply:
I understand. While I still believe we are using the incorrect term, I have changed it everywhere in e50eb11, we can table a discussion in another occasion.